Add links for ingresses

Nokel81 opened the pull request:
resolves: #4626

![Screen Shot 2022-01-05 at 11 34 01 AM](https://user-images.githubusercontent.com/8225332/148254311-489da9dd-344e-4575-9077-59722aaac7f2.png)
![Screen Shot 2022-01-05 at 12 16 36 PM](https://user-images.githubusercontent.com/8225332/148260253-dd45b4d6-047d-4b42-b57d-723e2f20b361.png)

Nokel81 committed:
Add links for ingresses

Signed-off-by: Sebastian Malton <sebastian@malton.name>

Nokel81 committed:
Add back the arrow service

Signed-off-by: Sebastian Malton <sebastian@malton.name>

nevalla commented on src/renderer/components/+network-ingresses/ingress-details.tsx:
```diff
@@ -65,44 +65,51 @@ export class IngressDetails extends React.Component<Props> {
+            {
+              rule.http.paths
+                .map(({ path = "/", backend }, index) => {
+                  const link = `${protocol}://${rule.host || "*"}${path}`;
```
if host is `*`, isn't hyperlink broken then?

Nokel81 replied:
Yes it is, I guess I should then not display them as links then.

Nokel81 committed:
Only display link if host is defined

Signed-off-by: Sebastian Malton <sebastian@malton.name>

jim-docker commented on src/renderer/components/+network-ingresses/ingresses.tsx:
```diff
@@ -55,7 +55,19 @@ export class Ingresses extends React.Component<IngressesProps> {
+                onClick={e => e.stopPropagation()}
+              >
+                {route.url}
+              </a> ⇢ {route.service}
```
This should also not be a link if `route.url` contains `*`?

jim-docker reviewed (commented):
doesn't work with resource backends? Just trying the example https://raw.githubusercontent.com/kubernetes/website/main/content/en/examples/service/networking/ingress-resource-backend.yaml from https://kubernetes.io/docs/concepts/services-networking/ingress/:

https://user-images.githubusercontent.com/40840436/158704799-a5083797-cf2c-45a5-9389-d02f9d31ad7a.mov

Nokel81 committed:
Resolve PR comments

- Fix crash in IngressDetails

- Make ingress routes scrollable

- Don't display link if the URL contains "*"

- Consolidate rendering of rules to all use the same transform function

Signed-off-by: Sebastian Malton <sebastian@malton.name>

jim-docker commented on src/common/k8s-api/endpoints/ingress.api.ts:
```diff
@@ -196,6 +198,24 @@ export class Ingress extends KubeObject {
 
+export function computeRuleDeclarations(ingress: Ingress, rule: IngressRule): ComputedIngressRoute[] {
+  const { host = "*", http: { paths } = { paths: [] }} = rule;
+  const protocol = (ingress.spec.tls?.length ?? 0) > 0
```
I'm seeing every link getting https, something fishy here?

jim-docker replied:
If I set `tls` it gets http so this is wrong, not exactly reversed with that `?? 0` I think

Nokel81 replied:
Thanks for catching this.

Nokel81 committed:
Fix http(s) formatting and add some tests

Signed-off-by: Sebastian Malton <sebastian@malton.name>
